llvm/llvm.nim

```diff
@@ -3,42 +3,75 @@
 # See the LICENSE file for license info (doh!)
 
 import strutils
+import strformat
+import os
+
+static:
+  if not (defined(linux) or defined(macosx)):
+    echo "Unsupported platform. Exiting."
+    quit(-1)
 
 const
-  LLVMLib = "libLLVM-14.so"
-  LLVMRoot = "../ext/llvm-14.0.0.src/"
-  LLDRoot = "../ext/lld-14.0.0.src/"
+  Root = currentSourcePath.parentDir().parentDir()
   LLVMVersion* = "14.0.0"
+  LLVMRoot = fmt"{Root}/ext/llvm-{LLVMVersion}.src"
+  LLDRoot = fmt"{Root}/ext/lld-{LLVMVersion}.src"
 
 {.passL: "-llldELF" .}
 {.passL: "-llldWasm" .}
 {.passL: "-llldMinGW" .}
 {.passL: "-llldCommon" .}
+{.passL: "-llldMachO" .}
 {.passL: "-lz" .}
+when defined(macosx): 
+  {.passL: "-lxar" .}
 
 when defined(staticLLVM):
   const
-    LLVMOut = LLVMRoot & "sta/"
+    LLVMOut = fmt"{LLVMRoot}/sta"
+
+  {.passL: gorge(fmt"{LLVMRoot}/sta/bin/llvm-config --libs all").}
 
-  {.passL: gorge(LLVMRoot & "sta/bin/llvm-config --libs all").}
+  when defined(macosx):
+    const SDKRoot = gorge("xcrun --sdk macosx --show-sdk-path")
+    {.passC: fmt"-isysroot{SDKRoot}".}
+    {.passL: fmt"-Wl,-syslibroot,{SDKRoot}".}
 
 else:
   const
-    LLVMOut = LLVMRoot & "sha/"
+    LLVMOut = fmt"{LLVMRoot}/sha"
+
+  when defined(macosx):
+    {.passL: "-lLLVM".}
+    {.passL: fmt"-Wl,-rpath,@loader_path/../ext/llvm-{LLVMVersion}.src/sha/lib".}
+
+  elif defined(linux):
+    {.passL: "-lLLVM-14".}
+    {.passL: fmt"-Wl,'-rpath=$ORIGIN/../ext/llvm-{LLVMVersion}.src/sha/lib'".}
+    
 
-  {.passL: "-lLLVM-14".}
-  {.passL: "-Wl,'-rpath=$ORIGIN/" & LLVMOut & "lib/'".}
+{.passC: fmt"-I{LLVMRoot}/include".}
+{.passC: fmt"-I{LLVMOut}/include".}
 
-{.passC: "-I" & LLVMRoot & "include".}
-{.passC: "-I" & LLVMOut & "include".}
+{.passC: fmt"-I{LLDRoot}/include".}
 
-{.passC: "-I" & LLDRoot & "include".}
+when defined(linux):
+  {.passL: "-Wl,--as-needed".}
+else:
+  {.passL: "-Wl".}
+
+{.passL: gorge(fmt"{LLVMOut}/bin/llvm-config --ldflags").}
+{.passL: gorge(fmt"{LLVMOut}/bin/llvm-config --system-libs").}
 
-{.passL: "-Wl,--as-needed".}
-{.passL: gorge(LLVMOut & "bin/llvm-config --ldflags").}
-{.passL: gorge(LLVMOut & "bin/llvm-config --system-libs").}
+{.compile("wrapper.cc", "-std=c++14").}
```

Comment:
and with a small upstream fix, this is not needed either: https://github.com/arnetheduck/nlvm/commit/435502228c3df81e6af86a698aec9d36567c7018